C/common/pipeline_branch.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Fledge pipeline branch class
+ *
+ * Copyright (c) 2024 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+#include <pipeline_element.h>
+#include <filter_pipeline.h>
+#include <config_handler.h>
+#include <service_handler.h>
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+
+using namespace std;
+
+/**
+ * Constructor for a branch in a filter pipeline
+ */
+PipelineBranch::PipelineBranch(FilterPipeline *parent) : m_pipeline(parent), m_thread(NULL), PipelineElement()
+{
+	m_shutdownCalled = false;
+}
+
+/**
+ * Destructor for the pipeline branch
+ *
+ * If the pipeline is not already shutdown then shut it down
+ * Delete the thread if it exists.
+ */
+PipelineBranch::~PipelineBranch()
+{
+	if (!m_shutdownCalled)
+	{
+		m_shutdownCalled = true;
+		m_cv.notify_all();
+		m_thread->join();
```

Comment:
Is it necessary to call _m_thread->joinable()_ before doing this?

Reply:
It should not be, but it is a good precaution to take